src/vma/proto/neighbour.cpp

```diff
@@ -405,9 +407,22 @@ bool neigh_entry::post_send_udp(iovec * iov, header *h)
 #endif
 	mem_buf_desc_t* p_mem_buf_desc, *tmp = NULL;
 	tx_packet_template_t *p_pkt;
-
+	int mtu = m_p_ring->get_mtu();
 	size_t sz_data_payload = iov->iov_len;
-	size_t max_ip_payload_size = ((m_p_ring->get_mtu() - sizeof(struct iphdr)) & ~0x7);
+	iphdr* p_hdr = &h->m_header.hdr.m_ip_hdr;
+	route_rule_table_key rtk(p_hdr->daddr, p_hdr->saddr, 0);
+	cache_entry_subject<route_rule_table_key, route_val*>* p_ces = NULL;
+	p_ces = g_p_route_table_mgr->get_entry(rtk);
```

Comment:
@rafiw pay attention that `get_entry` is not thread safe

Reply:
@igor-ivanov  i added locking
what do you think?